### PR TITLE
chakrashim: Use JsCreateExternalObjectWithPrototype

### PR DESCRIPTION
**Pull request by kfarnung**
Previously, Chakrashim was calling both JsCreateExternalObject and JsSetPrototype.
JsSetPrototype is expensive to lazy call. Use combined JsCreateExternalObjectWithPrototype instead

This depends on the changes in https://github.com/Microsoft/ChakraCore/pull/4719 landing first to prevent TTD breaks.

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Affected core subsystem(s)

chakrashim

### CODE REVIEW

**Comment by kfarnung** [comment]
I adopted #429, assuming @obastemur is ok with this, we can close his PR.

File: deps/chakrashim/src/v8objecttemplate.cc [hunk 1]
```diff
@@ -860,35 +860,35 @@ Local<Object> ObjectTemplate::NewInstance(Handle<Function> constructor) {
...
 
     if (JsGetProperty(*constructor,
                       iso->GetCachedPropertyIdRef(
                           jsrt::CachedPropertyIdRef::prototype),
-                      &prototypeValue) == JsNoError) {
-        if (JsSetPrototype(newInstanceRef, prototypeValue) != JsNoError) {
-          return Local<Object>();
-        }
+                      &prototype) != JsNoError) {
+      prototype = nullptr;
```
Comment (kfarnung) [inline deps/chakrashim/src/v8objecttemplate.cc:878]: I went back and forth on this, I largely think it's unnecessary, but I landed on the safe side.

**Comment by kfarnung** [comment]
CI: ~~https://ci.nodejs.org/job/chakracore-test-pull-request/200/~~
https://ci.nodejs.org/job/chakracore-test-pull-request/201/

**Review by boingoing — APPROVED** [review]
Looks right to me

**Comment by obastemur** [comment]
@kfarnung Thanks

**Pull request merged**